### PR TITLE
Fix up models module to get tests up to 100%

### PR DESCRIPTION
**Pull request by mdmurbach**

### CODE REVIEW

**Comment by mdmurbach** [comment]
Related to #96

File: impedance/models/circuits/circuits.py [hunk 1]
```diff
@@ -404,11 +403,13 @@ def __init__(self, CPE=False, **kwargs):
...
+        if len(self.initial_guess) + len(self.constants) != circuit_len:
+            raise ValueError('The number of initial guesses ' +
+                             f'({len(self.initial_guess)}) + ' +
+                             'the number of constants ' +
+                             f'({len(self.constants)})' +
+                             ' must be equal to ' +
+                             f'the circuit length ({circuit_len})')
 
 
 class CustomCircuit(BaseCircuit):
```
Comment (BGerwe) [inline impedance/models/circuits/circuits.py]: Sort of tangential, but can we make `initial_guess` an explicit keyword argument? If they're omitted an error is raised, so making it explicit could communicate the requirement better to the user.

Comment (mdmurbach) [reply impedance/models/circuits/circuits.py]: Challenge here is in loading a circuit from a .json file. It we want .load to be a method of BaseCircuit, then we need to be able to create a circuit instance (without an initial guess) and then call .load(filename). The alternative is a separate function (which is how it was until recently), but I thought it was neater to have it live with the circuit model class so I moved it in the refactor. 

To me, it makes sense to be able to create a circuit without an initial guess, but we have to make sure that we raise a meaningful error if the user tries to fit or predict before supplying one.

Comment (BGerwe) [reply impedance/models/circuits/circuits.py]: Got it. I'm Ok living with it as-is, then.

File: impedance/tests/test_circuit_elements.py [hunk 2]
```diff
@@ -46,20 +44,18 @@ def test_all():
             assert np.isclose(val, correct_vals[key]).all()
```
Comment (BGerwe) [inline impedance/tests/test_circuit_elements.py]: When I run this test in a Jupyter Lab instance I get an error. The first related to `'__builtins__'` in `circuit_elements`:
```
AttributeError: 'dict' object has no attribute 'num_params'
```
I don't get this same error running pytest locally so maybe Jupyter isn't handling the globals appropriately?

File: impedance/tests/test_circuit_elements.py [hunk 3]
```diff
@@ -46,20 +44,18 @@ def test_all():
...
             f(['hi'], ['yes', 'hello'])
-        except(AssertionError):
-            pass
-        else:
-            raise Exception('unhandled error occurred')
+
+    # Test no overflow in T at high frequencies
+    with pytest.warns(None) as record:
+        circuit_elements['T']([1, 2, 50, 100], [10000])
+    assert not record
```
Comment (BGerwe) [inline impedance/tests/test_circuit_elements.py]: I get an error here running in a Jupyter Lab instance and running pytest locally. I get `AssertionError` from both approaches. Any idea why my machine would overflow while yours doesn't?

Comment (mdmurbach) [reply impedance/tests/test_circuit_elements.py]: I think the warnings within the context of `with pytest.warns(None)` are stored in record. By asserting `not record`, the check is to make sure that no warnings were raised

Comment (BGerwe) [reply impedance/tests/test_circuit_elements.py]: Alright, then that's how I originally understood it. So I think that means my machine is giving overflow when yours isn't?

**Pull request merged**